dags/walg_backup.py

```diff
@@ -0,0 +1,112 @@
+from airflow.operators.python_operator import PythonOperator
+from dags.exec_in_pod import exec_in_pod
+
+import os
+import json
+
+YESTERDAY = datetime.now() - timedelta(days=1)
+
+namespace = os.getenv('NAMESPACE')
+in_cluster = os.getenv('LOCAL_AIRFLOW', False) == 'False'
+
+default_args = {
+    'owner': 'airflow',
+    'depends_on_past': False,
+    'start_date': YESTERDAY,
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': timedelta(minutes=5),
+}
+
+full_exec_command = [
+    '/bin/sh',
+    '-c',
+      'PGPASSWORD=$PGPASSWORD_SUPERUSER envdir $WALE_ENV_DIR wal-g backup-push $PGDATA'
+]
+
+incremental_exec_command = [
+    '/bin/sh',
+    '-c',
+    'lsn_string=$(psql -qtA -c \'select pg_walfile_name(pg_current_wal_lsn())\'); echo "$lsn_string"; lsn_number=$(echo $lsn_string | cut -c 9-); echo $lsn_number; PGPASSWORD=$PGPASSWORD_SUPERUSER envdir $WALE_ENV_DIR wal-g catchup-push $PGDATA --from-lsn 0x$lsn_number'
+]
+
+## Work for later: recovery dag
+
+# https://www.postgresql.org/docs/12/runtime-config-wal.html#RUNTIME-CONFIG-WAL-RECOVERY-TARGET
+# before doing this, replicas should be lowered to 1
+# full_restore_exec_command = [
+#     '/bin/sh',
+#     '-c',
+#     'patronictl pause; \\
+#     pg_ctl stop; \\
+#     envdir $WALE_ENV_DIR wal-g backup-fetch /tmp/walg LATEST; \\
+#     touch /tmp/walg/recovery.signal; \\
+#     pg_ctl -D /tmp/walg/ start -o "--restore_command=\'PGPASSWORD=$PGPASSWORD_SUPERUSER envdir $WALE_ENV_DIR wal-g wal-fetch %f %p\' --recovery_target_time=\'2020-06-11 11:09:00-07\' --recovery_target_action=promote"; \\
+#     pg_ctl -D /tmp/walg/ stop; \\
+#     mv $PGDATA $PGDATA-bak; \\
+#     mv /tmp/walg/ $PGDATA; \\
+#     pg_ctl start; \\
+#     patronictl resume; \\
+#     rm -rf $PGDATA-bak'
+# ]
+
+# incremental_restore_exec_command = [
+#     '/bin/sh',
+#     '-c',
+#     'patronictl pause; \\
+#     pg_ctl stop; \\
+#     envdir $WALE_ENV_DIR wal-g catchup-fetch /tmp/walg LATEST; \\
+#     touch /tmp/walg/recovery.signal; \\
+#     pg_ctl -D /tmp/walg/ start -o "--restore_command=\'PGPASSWORD=$PGPASSWORD_SUPERUSER envdir $WALE_ENV_DIR wal-g wal-fetch %f %p\' --recovery_target_time=\'2020-06-11 11:09:00-07\' --recovery_target_action=promote"; \\
+#     pg_ctl -D /tmp/walg/ stop; \\
+#     mv $PGDATA $PGDATA-bak; \\
+#     mv /tmp/walg/ $PGDATA; \\
+#     pg_ctl start; \\
+#     patronictl resume; \\
+#     rm -rf $PGDATA-bak'
+# ]
+
+
+def test(a,b,c):
+  print('hello')
+
+DAG_ID = os.path.basename(__file__).replace(".pyc", "").replace(".py", "")
+SCHEDULE_INTERVAL = '@hourly'
+
+ciip_incremental_backup = DAG(DAG_ID + '_ciip_incremental', default_args=default_args, schedule_interval=SCHEDULE_INTERVAL)
+ciip_full_backup = DAG(DAG_ID + '_ciip_full', default_args=default_args, schedule_interval=None)
+
+ggircs_incremental_backup = DAG(DAG_ID + '_ggircs_incremental', default_args=default_args, schedule_interval=SCHEDULE_INTERVAL)
```

Comment:
ggircs can be backed up daily, as we won't make more that one change a day hopefully.
We might want to trigger a specific time, and sync with the ECCC incremental swrs import (trigger the backup an hour before the swrs import)